app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,57 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.database.FeedDao
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedItemDto
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.schedulers.Schedulers
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+class FeedRepository(
+    private val feedApiService: FeedApiService,
+    private val feedDao: FeedDao
+):Repository
+{
+    override fun getFeed(): Observable<List<FeedItem>> {
+        return feedDao.getAll()
+            .subscribeOn(Schedulers.io())
+            .map { it.toFeedItems() }
+            .onErrorReturn { emptyList() }
```

Comment:
We did talk about how  since I assume on my view model that this function cannot fail that I should deal with it here. So i thought it would be nice to just sent an empty list whenever there is an error.